Add FPM proxy script.

ankitrox opened the pull request:
## Summary

<!-- Please reference the issue this PR addresses in the following list. -->
Addresses issue:

- #9665 

## Relevant technical choices

<!-- Please describe your changes. -->

## PR Author Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 5.2 and PHP 7.4.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [x] Run the code.
- [x] Ensure the acceptance criteria are satisfied.
- [x] Reassess the implementation with the IB.
- [x] Ensure no unrelated changes are included.
- [x] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [x] Ensure there is a QA Brief.
- [x] Ensure there are no unexpected significant changes to file sizes.

## Merge Reviewer Checklist

- [x] Ensure the PR has the correct target branch.
- [x] Double-check that the PR is okay to be merged.
- [x] Ensure the corresponding issue has a ZenHub release assigned.
- [x] Add a changelog message to the issue.

techanvil commented on fpm/measurement.php:
Although you've added an exclusion for this file to `phpcs.xml`, the formatting has still been modified when it's been saved.

We should keep the changes to the original to the absolute minimum to make life easier when updating it to a later version.

Please can you ensure the file is including with its original formatting? The only diff should be the addition of these lines:

```php
 * NOTICE: This file has been modified from its original version in accordance with the Apache License, Version 2.0.
```

```php
if ( isset( $_GET['healthCheck'] ) ) { // phpcs:ignore WordPress.Security.NonceVerification
	echo 'ok';
	exit;
}

```

You might also want to include the following line at the top of the file to make it easier to modify in editors like VSCode:

```php
<?php // phpcs:disable ?>
```

techanvil reviewed (changes requested):
Hi @ankitrox, this is most of the way there. Please see my comment about keeping the diff to the original `measurement.php` to a minimum.

Also, regarding the QAB:

- It would be worth adding a bit more detail to the point about checking the JS code is loaded on the page.
- It's also worth adding some info about confirming Analytics events are tracked as per the AC - neither the Analytics Debugger extension or Tag Manager currently work with the proxy script, @hussain-t and I had to use the realtime Analytics report to verify that events were indeed being tracked.

ankitrox commented:
Thanks @techanvil for reviewing the PR.

I have updated the proxy script to the one provided by Evan in this [slack thread](https://fueled10up.slack.com/archives/C077DB69US2/p1731602593260239) so that it will be easy to update with minimal changes.

Also, in the QAB I have added the point to test whether the events are being captured. This cannot be done straight away and will require update to Gtag file (until we have the functioning toggle for the FPM), so I have added steps for it in QAB, but at the same time added a link to test site where this modification is already done so that it will be easier for QAs to test and they will also know what has been changed and where.

Please let me know if this looks good.

Thanks again.

techanvil commented:
> Thanks @techanvil for reviewing the PR.
> 
> I have updated the proxy script to the one provided by Evan in this [slack thread](https://fueled10up.slack.com/archives/C077DB69US2/p1731602593260239) so that it will be easy to update with minimal changes.
> 
> Also, in the QAB I have added the point to test whether the events are being captured. This cannot be done straight away and will require update to Gtag file (until we have the functioning toggle for the FPM), so I have added steps for it in QAB, but at the same time added a link to test site where this modification is already done so that it will be easier for QAs to test and they will also know what has been changed and where.
> 
> Please let me know if this looks good.
> 
> Thanks again.

Thanks @ankitrox. That's generally LGTM, I have however pushed a small change to update the comments in `measurement.php` in order to keep the diff between that and the original file to a minimum.

This implies the site you've deployed the change to would need an update, however, as discussed on [Slack](https://fueled10up.slack.com/archives/C077DB69US2/p1732633909653789?thread_ts=1732628245.087959&cid=C077DB69US2) it's generally preferable for a QA tester to deploy the plugin themselves. Furthermore QA should be done against the `develop` or `main` branch once the PR has been merged, your site would presumably have had the branch build deployed rather than one of the mainline branches.

So, what I've done is simply remove the point about testing on that site from the QAB and specify that the QA should be done by an engineer, I think that's the best way forward for this one.

techanvil reviewed (approved):
LGTM, thanks @ankitrox!
